src/main/scala/io/reactivesocket/tck/DSLTestSubscriber.scala

```diff
@@ -25,15 +25,15 @@ import org.json4s.native.Serialization
 
 
 class DSLTestSubscriber(writer : PrintWriter, initData: String, initMeta: String, kind: String,
-  client: DSLTestClient = null) extends Subscriber[Payload] with Subscription {
+  client: Option[DSLTestClient]) extends Subscriber[Payload] with Subscription {
 
   implicit val formats = Serialization.formats(NoTypeHints)
 
   private var id: UUID = null
   this.id = UUID.randomUUID
 
   private var clientID: Int =
```

Comment:
I think this can just be

= client.map.(_.id).getOrElse(ClientIDGen.getNewID())